@andydotxyz It's failing the tests.

```
jezek@zen ~/projects/programming/x-go-bindings/xgbutil ‹fix/reuse-ids*› 
$ go test ./...                                                                                                                          [19:31:37]
# github.com/jezek/xgbutil
./xgbutil.go:279:11: xu.conn.SetIDRangeFunc undefined (type *xgb.Conn has no field or method SetIDRangeFunc)
FAIL	github.com/jezek/xgbutil [build failed]
FAIL	github.com/jezek/xgbutil/ewmh [build failed]
?   	github.com/jezek/xgbutil/gopher	[no test files]
FAIL	github.com/jezek/xgbutil/icccm [build failed]
FAIL	github.com/jezek/xgbutil/keybind [build failed]
FAIL	github.com/jezek/xgbutil/motif [build failed]
FAIL	github.com/jezek/xgbutil/mousebind [build failed]
FAIL	github.com/jezek/xgbutil/xcursor [build failed]
FAIL	github.com/jezek/xgbutil/xevent [build failed]
FAIL	github.com/jezek/xgbutil/xgraphics [build failed]
FAIL	github.com/jezek/xgbutil/xinerama [build failed]
FAIL	github.com/jezek/xgbutil/xprop [build failed]
?   	github.com/jezek/xgbutil/xrect	[no test files]
FAIL	github.com/jezek/xgbutil/xwindow [build failed]
FAIL
1 ↵                                                                                                                                 0.3s [19:31:37] 
```

--- a/go.mod
+++ b/go.mod
@@ -5,5 +5,5 @@ go 1.11
 require (
 	github.com/BurntSushi/freetype-go v0.0.0-20160129220410-b763ddbfe298
 	github.com/BurntSushi/graphics-go v0.0.0-20160129215708-b43f31a4a966
-	github.com/jezek/xgb v1.2.1-0.20260119194513-46ea16ee069c
+	github.com/jezek/xgb v1.3.0
 )

--- a/go.mod
+++ b/go.mod
@@ -5,5 +5,5 @@ go 1.11
 require (
 	github.com/BurntSushi/freetype-go v0.0.0-20160129220410-b763ddbfe298
 	github.com/BurntSushi/graphics-go v0.0.0-20160129215708-b43f31a4a966
-	github.com/jezek/xgb v1.1.0
+	github.com/jezek/xgb v1.2.1-0.20260119194513-46ea16ee069c
 )

--- a/xgbutil.go
+++ b/xgbutil.go
@@ -274,6 +274,7 @@ func NewConnXgb(c *xgb.Conn) (*XUtil, error) {
 			"because the XINERAMA extension could not be loaded.")
 	}
 
+	// initialise the XCMisc extension for ID re-use - if successful then hook in to xgb.
 	err = xcmisc.Init(c)
 	if err == nil {
 		xu.conn.SetIDRangeFunc(func(c *xgb.Conn) (uint32, uint32, error) {

